rohmu/encryptor.py

```diff
@@ -34,21 +30,33 @@ class EncryptorError(Exception):
 
 
 class Encryptor:
-    def __init__(self, rsa_public_key_pem):
+    def __init__(self, rsa_public_key_pem: Union[str, bytes]):
         if not isinstance(rsa_public_key_pem, bytes):
             rsa_public_key_pem = rsa_public_key_pem.encode("ascii")
-        self.rsa_public_key = serialization.load_pem_public_key(rsa_public_key_pem, backend=default_backend())
-        self.cipher = None
-        self.authenticator = None
+        self.rsa_public_key = cast(
+            RSAPublicKey, serialization.load_pem_public_key(rsa_public_key_pem, backend=default_backend())
+        )
+        self._cipher: Optional[CipherContext] = None
+        self._authenticator: Optional[HMAC] = None
+
+    @property
+    def cipher(self) -> CipherContext:
+        assert self._cipher is not None
```

Comment:
nit: Wouldn't be better to raise a proper exception here instead of using `assert`? (like you did in https://github.com/aiven/rohmu/pull/104/files#diff-5c190a4f634c25382cbcfbc9997d46e5668951c55b395c4082beb4c7e3077312R92)

Reply:
I decided to go with a custom `UninitializedException`.